frontend/src/pages/PipelineDetailsV2.tsx

```diff
@@ -122,7 +123,7 @@ function PipelineDetailsV2({
       {selectedTab === 1 && (
         <div className={commonCss.codeEditor} data-testid={'spec-ir'}>
           <Editor
-            value={JSON.stringify(JSON.parse(templateString || ''), null, 2)}
+            value={jsyaml.safeDump(jsyaml.safeLoad(templateString || ''))}
```

Comment:
Why do we load and dump here? It is equivalent to `templateString || ''` directly.

Reply:
I just follow the JSON logic to make the changes. You are correct. I can directly use the string! Thank you @zijianjoy !